lib/decorateWithMetrics.js

```diff
@@ -136,55 +137,83 @@ function toDuration(millis, raw = false) {
-                endTime: toDuration(call.endTime, raw),
-                duration: toDuration(call.duration, raw),
-                args: JSON.stringify(call.args)
-            }));
-
-            return {
+        .map(({methodName, calls, ...methodStats}) => {
+            tableSummary.addRow(
                 methodName,
-                ...prettyTimes,
-                'time last call completed': toDuration(lastCall.endTime, raw),
-                calls: calls.length,
-                prettyCalls,
-            };
+                toDuration(methodStats.total, raw),
+                toDuration(methodStats.max, raw),
+                toDuration(methodStats.min, raw),
+                toDuration(methodStats.avg, raw),
+                toDuration(methodStats.lastCall.endTime, raw),
+                calls.length,
+            );
+
+            const callsTable = MDTable.factory({
+                title: methodName,
+                heading: ['start time', 'end time', 'duration', 'args'],
+                leftAlignedCols: [3],
+                rows: calls.map(call => ([
+                    toDuration(call.startTime, raw),
+                    toDuration(call.endTime, raw),
+                    toDuration(call.duration, raw),
+                    call.args.map(String).join(', ').slice(0, 60), // Restrict cell width to 60 chars max
+                ]))
+            });
+
+            return callsTable;
         })
         .value();
 
-    /* eslint-disable no-console */
-    console.group('Onyx Benchmark');
-    console.info('  Total: ', toDuration(totalTime, raw));
-    console.info('  Last call finished at: ', toDuration(lastCompleteCall.endTime, raw));
+    if (/csv|json|string/i.test(format)) {
+        const allTables = [tableSummary, ...methodCallTables];
+
+        return allTables.map((table) => {
+            switch (format.toLowerCase()) {
+                case 'csv':
+                    return table.toCSV();
+                case 'json':
+                    return table.toJSON();
+                default:
+                    return table.toString();
+            }
+        }).join('\n\n');
+    }
 
-    console.table(prettyData.map(({prettyCalls, ...summary}) => summary));
+    const mainOutput = [
+        '### Onyx Benchmark',
+        `  - Total: ${toDuration(totalTime, raw)}`,
+        `  - Last call finished at: ${lastCompleteCall ? toDuration(lastCompleteCall.endTime, raw) : 'N/A'}`,
+        '',
+        tableSummary.toString()
+    ];
 
-    prettyData.forEach((method) => {
-        console.groupCollapsed(`[${method.methodName}] individual calls: `);
-        console.table(method.prettyCalls);
+    /* eslint-disable no-console */
+    console.info(mainOutput.join('\n'));
+    methodCallTables.forEach((table) => {
+        console.groupCollapsed(table.getTitle());
+        console.info(table.toString());
         console.groupEnd();
     });
-
-    console.groupEnd();
```

Comment:
I wonder whether we need console printing at all 
The only convenience here is `console.groupCollapsed`